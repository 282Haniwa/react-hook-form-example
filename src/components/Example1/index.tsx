import React from 'react'

// eslint-disable-next-line import/no-unresolved
import example1WithRHFCode from '!!raw-loader!src/components/Example1/Example1WithRHF.tsx'
// eslint-disable-next-line import/no-unresolved
import example1WithStateCode from '!!raw-loader!src/components/Example1/Example1WithState.tsx'
import { Example } from 'src/components/commons/Example'

import { Example1WithState } from 'src/components/Example1/Example1WithState'
import { Example1WithRHF } from 'src/components/Example1/Example1WithRHF'

type Props = {}

export const Example1: React.FC<Props> = () => {
  return (
    <Example
      withState={{
        sample: <Example1WithState />,
        code: example1WithStateCode,
      }}
      withRHF={{
        sample: <Example1WithRHF />,
        code: example1WithRHFCode,
      }}
    />
  )
}
