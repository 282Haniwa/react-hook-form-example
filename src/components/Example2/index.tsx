import React from 'react'

// eslint-disable-next-line import/no-unresolved
import Example2WithRHFCode from '!!raw-loader!src/components/Example2/Example2WithRHF.tsx'
// eslint-disable-next-line import/no-unresolved
import Example2WithStateCode from '!!raw-loader!src/components/Example2/Example2WithState.tsx'
import { Example } from 'src/components/commons/Example'

import { Example2WithState } from 'src/components/Example2/Example2WithState'
import { Example2WithRHF } from 'src/components/Example2/Example2WithRHF'

type Props = {}

export const Example2: React.FC<Props> = () => {
  return (
    <Example
      left={{
        label: '従来の実装',
        sample: <Example2WithState />,
        code: Example2WithStateCode,
      }}
      right={{
        label: 'react-hook-formを使った実装',
        sample: <Example2WithRHF />,
        code: Example2WithRHFCode,
      }}
    >
      バリデーションありのフォームの実装です。
      <br />
      基本的なバリデーションに関してはロジックが用意してあるので、圧倒的に記述量が減ります。
      <br />
      値を引数にとってエラーメッセージを返り値とするカスタムバリデーション関数をregisterで設定することも可能です。
      <br />
      <br />
      また、useFormのmodeプロパティを設定することで、バリデーションの走るタイミングを制御することも可能です。
      <br />
      <a href="https://react-hook-form.com/api#useForm">
        https://react-hook-form.com/api#useForm
      </a>
    </Example>
  )
}
