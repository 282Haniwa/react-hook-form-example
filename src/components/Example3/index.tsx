import React from 'react'

// eslint-disable-next-line import/no-unresolved
import Example3WithRHFCode from '!!raw-loader!src/components/Example3/Example3WithRHF.tsx'
// eslint-disable-next-line import/no-unresolved
import Example3WithStateCode from '!!raw-loader!src/components/Example3/Example3WithState.tsx'
import { Example } from 'src/components/commons/Example'

import { Example3WithState } from 'src/components/Example3/Example3WithState'
import { Example3WithRHF } from 'src/components/Example3/Example3WithRHF'

type Props = {}

export const Example3: React.FC<Props> = () => {
  return (
    <Example
      withState={{
        sample: <Example3WithState />,
        code: Example3WithStateCode,
      }}
      withRHF={{
        sample: <Example3WithRHF />,
        code: Example3WithRHFCode,
      }}
    >
      react-hook-formは入力値の状態をstateで管理していないので、入力値が更新されるたびにレンダリングが走るということがありません。
      <br />
      これによって、コンポーネントの再描画のコストが減ります。
    </Example>
  )
}
