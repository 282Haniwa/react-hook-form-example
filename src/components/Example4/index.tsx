import React from 'react'

// eslint-disable-next-line import/no-unresolved
import Example4WithRHFCode from '!!raw-loader!src/components/Example4/Example4WithRHF.tsx'
// eslint-disable-next-line import/no-unresolved
import Example4WithStateCode from '!!raw-loader!src/components/Example4/Example4WithState.tsx'
import { Example } from 'src/components/commons/Example'

import { Example4WithState } from 'src/components/Example4/Example4WithState'
import { Example4WithRHF } from 'src/components/Example4/Example4WithRHF'

type Props = {}

export const Example4: React.FC<Props> = () => {
  return (
    <Example
      left={{
        label: '従来の実装',
        sample: <Example4WithState />,
        code: Example4WithStateCode,
      }}
      right={{
        label: 'react-hook-formを使った実装',
        sample: <Example4WithRHF />,
        code: Example4WithRHFCode,
      }}
    >
      フィールドのリストを実装するサンプルです。
      <br />
      ほとんどの配列操作のメソッドが定義されているので、値の配列をstateで管理してそれを操作するメソッドを実装する必要がありません。
      <br />
      <a href="https://react-hook-form.com/api#useFieldArray">
        https://react-hook-form.com/api#useFieldArray
      </a>
    </Example>
  )
}
