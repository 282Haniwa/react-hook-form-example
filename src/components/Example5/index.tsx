import React from 'react'

// eslint-disable-next-line import/no-unresolved
import Example5WithRHFCode from '!!raw-loader!src/components/Example5/Example5WithRHF.tsx'
// eslint-disable-next-line import/no-unresolved
import TextFieldCode from '!!raw-loader!src/components/commons/TextField/TextField.tsx'
import { Example } from 'src/components/commons/Example'

import { Example5WithRHF } from 'src/components/Example5/Example5WithRHF'

type Props = {}

export const Example5: React.FC<Props> = () => {
  return (
    <Example
      withState={{
        // label: '例',
        sample: <Example5WithRHF />,
        code: Example5WithRHFCode,
      }}
      withRHF={{
        // label: 'TextFieldコンポーネント',
        sample: '',
        code: TextFieldCode,
      }}
    >
      TextFieldコンポーネントで実装したサンプルです。
    </Example>
  )
}
