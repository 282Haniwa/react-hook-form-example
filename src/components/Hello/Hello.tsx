import React from 'react'

type Props = {
  name: string
}

export const Hello: React.FC<Props> = (props) => {
  const { name } = props
  return <div>Hello {name}!</div>
}
