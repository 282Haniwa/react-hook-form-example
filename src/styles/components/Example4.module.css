.field-list {
  padding: 0;
  margin: 0;
}

.field-item:not(:last-child) {
  padding: 0;
  margin-bottom: 16px;
}

.field {
  min-width: 60px;
  margin-right: 8px;
  flex-grow: 1;
}
