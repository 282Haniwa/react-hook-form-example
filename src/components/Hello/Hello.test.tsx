import React from 'react'
import { shallow } from 'enzyme'
import { Hello } from 'src/components/Hello'

test('Hello test', () => {
  const hello = shallow(<Hello name="World" />)

  // Interaction demo
  expect(hello.text()).toEqual('Hello World!')
  hello.setProps({ name: 'Jest' })
  expect(hello.text()).toEqual('Hello Jest!')

  // Snapshot demo
  expect(hello).toMatchSnapshot()
})
