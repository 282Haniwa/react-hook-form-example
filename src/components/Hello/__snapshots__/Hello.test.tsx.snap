// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello test 1`] = `
<div>
  Hello 
  Jest
  !
</div>
`;
